tests/common/system_utils/docker.py

```diff
@@ -0,0 +1,156 @@
+
+_DockerRegistryInfo = collections.namedtuple('DockerRegistryInfo', 'host username password')
+class DockerRegistryInfo(_DockerRegistryInfo):
+    """
+        DockerRegistryInfo holds all the data needed to access a remote Docker registry.
+
+        Attributes:
+            host (str): The remote host where the Docker registry is located.
+            username (str): The username used to access the registry.
+            password (str): The password used to access the registry.
+    """
+    pass
+
+def parse_registry_file(registry_file):
+    """
+        parse_registry_file parses the provided file to produce a DockerRegistryInfo.
+
+        See `SONIC_DOCKER_REGISTRY` for the expected format of this file.
+
+        Args:
+            registry_file (str): The name of the file holding the registry information.
+
+        Raises:
+            IOError: If the file cannot be opened for any reason.
+            ValueError: If the provided file is missing any required fields.
+
+        Returns:
+            DockerRegistryInfo: The registry info from the registry file.
+    """
+
+    try:
+        with open(registry_file) as contents:
+            registry_vars = yaml.safe_load(contents)
+    except IOError as err:
+        _LOGGER.error("Failed to parse registry file \"%s\" (%s)", registry_file, err)
+        raise
+
+    host = registry_vars.get("docker_registry_host")
+    username = registry_vars.get("docker_registry_username")
+    password = registry_vars.get("docker_registry_password")
+
+    if not host or not username or not password:
+        error_message = "Registry file \"{}\" is missing login or hostname".format(registry_file)
+        _LOGGER.error(error_message)
+        raise ValueError(error_message)
+
+    return DockerRegistryInfo(host, username, password)
+
+def delete_container(dut, container_name):
+    """
+        delete_container attempts to delete the specified container from the DUT.
+
+        Args:
+            dut (SonicHost): The target device.
+            container_name (str): The name of the container to delete.
+    """
+
+    dut.docker_container(name=container_name, state="absent")
+
+def download_image(dut, registry, image_name, image_version="latest"):
+    """
+        download_image attempts to download the specified image from the registry.
+
+        Args:
+            dut (SonicHost): The target device.
+            registry (DockerRegistryInfo): The registry from which to pull the image.
+            image_name (str): The name of the image to download.
+            image_version (str): The version of the image to download.
+    """
+
+    dut.docker_login(registry_url=registry.host,
+                     username=registry.username,
+                     password=registry.password)
+    dut.docker_image(source="pull",
+                     name="{}/{}:{}".format(registry.host, image_name, image_version))
+
+def tag_image(dut, tag, image_name, image_version="latest"):
```

Comment:
these are very slim wrappers around Ansible APIs. You can use the Ansible API directly.

Reply:
I kept the wrappers because the interface is a lot simpler than the Ansible APIs. The Ansible APIs are very useful for this sort of thing, but they're also very flexible and have dozens of options to sift through. Most of the time, we really only need to use a few of them.

This just saves some effort digging through the Ansible docs to find which 3 options you need (plus, if someone /really/ needs to go do something super specific, they can still use the Ansible API directly).